meteor/client/ui/SegmentTimeline/Renderers/VTSourceRenderer.tsx

```diff
@@ -86,41 +86,103 @@ export class VTSourceRendererBase extends CustomLayerItemRenderer<IProps & WithT
 		}
 	}
 
+	mountRightLabelContainer(
+		props: IProps,
+		prevProps: IProps | null,
+		newState: Partial<IState>,
+		itemElement: HTMLElement | null
+	): Partial<IState> {
+		if (this.rightLabelContainer && itemElement) {
+			const itemDuration = this.getItemDuration(true)
+			if (prevProps === null || itemElement !== prevProps.itemElement) {
+				if (itemDuration === Number.POSITIVE_INFINITY) {
+					itemElement.parentNode &&
+						itemElement.parentNode.parentNode &&
+						itemElement.parentNode.parentNode.parentNode &&
+						itemElement.parentNode.parentNode.parentNode.appendChild(this.rightLabelContainer)
+
+					newState.rightLabelIsAppendage = true
+				} else {
+					this.rightLabelContainer?.remove()
+					itemElement.appendChild(this.rightLabelContainer)
+					newState.rightLabelIsAppendage = false
+				}
+			} else if (prevProps && prevProps.partDuration !== props.partDuration) {
+				if (itemDuration === Number.POSITIVE_INFINITY && this.state.rightLabelIsAppendage !== true) {
+					itemElement.parentNode &&
```

Comment:
With our adoption of newer TypeScript, I guess this can be written shorter now? `itemElement.parentNode?.parentNode` and so on?

Reply:
Same goes for the similar block above

Reply:
Yeah, I didn't notice that this can be now refactored. Good find.